src/Microsoft.TemplateEngine.Edge/Template/InputParameter.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.TemplateEngine.Edge.Template
+{
+    public class InputParameter
+    {
+        public InputParameter(string name, string? value)
+        {
+            Name = name;
```

Comment:
Consider using  `ITemplateParameter` instead. It may simplify a lot of logic and may make it easier to use in `IParameterSet`.